java/org/apache/catalina/connector/Request.java

```diff
@@ -2288,12 +2288,7 @@ public String getServletPath() {
      */
     @Override
     public HttpSession getSession() {
-        Session session = doGetSession(true);
-        if (session == null) {
-            return null;
-        }
-
-        return session.getSession();
+        getSession(true);
```

Comment:
No results returned.
```suggestion
        return getSession(true);
```

Reply:
Oh, I made a mistake!